egs/librispeech/WSASR/README.md

```diff
@@ -0,0 +1,205 @@
+  --words-file "${lang_dir}/words.txt" \
+  --sub-error-rate "${sub_er}" \
+  --ins-error-rate "${ins_er}" \
+  --del-error-rate "${del_er}" \
+  --output-cutset "${manifest_dir}/${synthetic_train_manifest}"
+```
+This script generates synthetic substitution, insertion, and deletion errors in the transcript with ratios 'sub_er', 'ins_er', and 'del_er', respectively. The original transcript is saved as 'verbatim transcript' in the cutset, along with information on how the transcript is corrupted:
+  - '[hello]' indicates the original word 'hello' is substituted by another word
+  - '[]' indicates an extra word is inserted into the transcript
+  - '-hello-' indicates the word 'hello' is deleted from the transcript
+So if the original transcript is "have a nice day" and the synthetic one is "a very good day", the 'verbatim transcript' would be:
+```
+original:  have  a      nice  day
+synthetic:       a very good  day
+verbatim: -have- a  [] [nice] day
+```
+
+### Training
+The training uses synthetic data based on the train-clean-100 subset.
+```
+otc_lang_dir=data/lang_bpe_200
+
+allow_bypass_arc=true
+allow_self_loop_arc=true
+initial_bypass_weight=-19
+initial_self_loop_weight=3.75
+bypass_weight_decay=0.975
+self_loop_weight_decay=0.999
+
+show_alignment=true
+
+export CUDA_VISIBLE_DEVICES="0,1,2,3"
+./conformer_ctc2/train.py \
+  --world-size 4 \
+  --manifest-dir "${manifest_dir}" \
+  --train-manifest "${synthetic_train_manifest}" \
+  --exp-dir "${exp_dir}" \
+  --lang-dir "${otc_lang_dir}" \
+  --otc-token "${otc_token}" \
+  --allow-bypass-arc "${allow_bypass_arc}" \
+  --allow-self-loop-arc "${allow_self_loop_arc}" \
+  --initial-bypass-weight "${initial_bypass_weight}" \
+  --initial-self-loop-weight "${initial_self_loop_weight}" \
+  --bypass-weight-decay "${bypass_weight_decay}" \
+  --self-loop-weight-decay "${self_loop_weight_decay}" \
+  --show-alignment "${show_alignment}"
+```
+The bypass arc deals with substitution and insertion errors, while the self-loop arc deals with deletion errors. Using "--show-alignment" would print the best alignment during training, which is very helpful for tuning hyperparameters and debugging.
+
+### Decoding
+```
+export CUDA_VISIBLE_DEVICES="0"
+./conformer_ctc2/decode.py \
+  --manifest-dir "${manifest_dir}" \
+  --exp-dir "${exp_dir}" \
+  --lang-dir "${otc_lang_dir}" \
+  --lm-dir "${lm_dir}" \
+  --otc-token "${otc_token}"
+```
+
+### Results (ctc-greedy-search)
+<table>
+  <tr>
+    <td rowspan=2>Training Criterion</td>
+    <td colspan=2>ssl</td>
+    <td colspan=2>fbank</td>
+  </tr>
+  <tr>
+    <td>test-clean</td>
+    <td>test-other</td>
+    <td>test-clean</td>
+    <td>test-other</td>
+  </tr>
+  <tr>
+    <td>CTC</td>
+    <td>100.0</td>
+    <td>100.0</td>
+    <td>99.89</td>
+    <td>99.98</td>
+  </tr>
+  <tr>
+    <td>OTC</td>
+    <td>11.89</td>
+    <td>25.46</td>
+    <td>20.14</td>
+    <td>44.24</td>
+  </tr>
+</table>
+
+### Results (1best, blank_bias=-4)
+<table>
```

Comment:
Could you upload your pretrained models to huggingface so that others can use the pretrained models with decode.py to reproduce your results?